library/Graphite/Grapher.php

```diff
@@ -19,12 +19,18 @@ class Grapher extends GrapherHook
     protected $baseUrl = 'http://graphite.com/render/?';
     protected $serviceMacro = 'icinga2.$host.name$.services.$service.name$.$service.check_command$.perfdata.$metric$.value';
     protected $hostMacro = 'icinga2.$host.name$.host.$host.check_command$.perfdata.$metric$.value';
-    protected $imageUrlMacro = '&target=$target$&source=0&width=300&height=120&hideAxes=true&lineWidth=2&hideLegend=true&colorList=049BAF';
-    protected $largeImageUrlMacro = '&target=$target$&source=0&width=800&height=700&colorList=049BAF&lineMode=connected';
+    protected $imageUrlMacro = '&target=$target$&source=0&width=300&height=120&hideAxes=true&lineWidth=2&hideLegend=true&colorList=$colorList$&areaMode=$areaMode$&areaAlpha=$areaAlpha$';
+    protected $largeImageUrlMacro = '&target=$target$&source=0&width=800&height=700&colorList=$colorList$&lineMode=connected&areaMode=$areaMode$&areaAlpha=$areaAlpha$';
+    protected $DerivativeMacro = 'summarize(nonNegativeDerivative($target$),\'$summarizeInterval$\', \'$summarizeFunc$\')';
     protected $legacyMode = false;
     protected $graphiteKeys = array();
     protected $graphiteLabels = array();
     protected $areaMode = "all";
+    protected $graphType = "normal";
+    protected $summarizeInterval = "10min";
+    protected $summarizeFunc = "sum";
+    protected $areaAlpha = "0.1";
+    protected $colorList = "049BAF,EE1D00,04B06E,0446B0,871E10,CB315D,B06904,B0049C";
```

Comment:
I think it would be better to avoid having properties for all the parameters e.g. the colorlist, areaalpha etc. otherwise this list will grow and grow. Perhaps if possible just add the graphite args to the template definitions?